src/mechanics/integrands/Base.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <memory>
+
+namespace NuTo
+{
+namespace Integrand
+{
+class Base
+{
+public:
+    virtual std::unique_ptr<Base> Clone() const = 0;
+    virtual ~Base() = default;
+
+    template <typename TIntegrand>
+    TIntegrand& As()
+    {
+        return dynamic_cast<TIntegrand&>(*this);
```

Comment:
Why is that now a dynamic_cast, and not done via the boost::any approach?

Reply:
I somehow thought that I need a polymorphic clone to fill the vector of integrands at each integration point in `Cell`. I have to do some more experimenting, but I might have been wrong. I could maybe just use a `std::vector<boost::any>` (value type( = nice!)) and be done. That would indeed be much better.

Reply:
I may be nice to have a common base class for other common methods like visualization. And if we have a common base class, there is no need to use boost::any.

Reply:
The `boost::any` topic is also content of the _neat ideas_ in [Project PDE](https://github.com/nutofem/nuto/projects/1).